### PR TITLE
RHIROS-388 - fixed store already initialized error

### PR DESCRIPTION
**Pull request by PreetiW**
Background:

We were facing 'store already initliazed' error on ROS service page while switching between ROS service and other services. This PR fixes the issue.

Jira Card: https://issues.redhat.com/browse/RHIROS-388

### CODE REVIEW

File: src/store/index.js [hunk 1]
```diff
@@ -1,25 +1,13 @@
-import ReducerRegistry from '@redhat-cloud-services/frontend-components-utilities/ReducerRegistry';
+import { getRegistry } from '@redhat-cloud-services/frontend-components-utilities/Registry';
 import promiseMiddleware from 'redux-promise-middleware';
 let registry;
 
 export function init (...middleware) {
-    if (registry) {
-        throw new Error('store already initialized');
-    }
-
```
Comment (PreetiW) [inline src/store/index.js:9]: @karelhala As we were facing this error while switching between ROS service and other services due to the above check, I updated the code to use the `getRegistory` method from `frontend-component-utitlies` to get access to the reducerRegistry object which already has the check around the registry. I also observed we are following the similar pattern in the `frontend-starter-app`.

I wanted to know though if any other services had faced similar issue in the past? do you have any insights to share around why we had decided to go with the above logic in the beginning. I am just curious about it. Thanks in advance!

cc: @kgaikwad

Comment (karelhala) [reply src/store/index.js:9]: Sure thing, so the problem is that when you switch from one app to another your app effectively re-runs the init function, 
which already is defined. Previously when app was always reloaded this was no issue as everything wss cleared out of memory and loaded once again.

Recently we've stopped sharing redux trough chrome as well which surfaced this problem. Previously the store was initialized and "stored" in chrome context, now it is stored within app itself so the app again tries to recreate the redux store and breaks with store already initialized. That's why the `getRegistry` was introduced a while back.

If you want to be completely certain that none other app will mess with your redux store I recoment using `React.createContext` and passing the registry in said context. Moreover you could create multiple stores and use them in various places as you want. An example of such usage can be seen in inventory - https://github.com/RedHatInsights/insights-inventory-frontend/blob/master/src/AppEntry.js#L12

**Review by karelhala — COMMENTED** [review]
Just one thing - when creating store use shortcutting, that way the production code won't blow up. Overall looks good!

File: src/AppEntry.js [hunk 2]
```diff
@@ -4,9 +4,12 @@ import { Provider } from 'react-redux';
 import { init } from './store';
 import App from './App';
 import { getBaseName } from '@redhat-cloud-services/frontend-components-utilities/helpers';
+import logger from 'redux-logger';
 
 const RosApp = () => (
-    <Provider store={ init().getStore() }>
+    <Provider store={ init(
+        ...(process.env.NODE_ENV !== 'production' ? [logger] : [])
+    ).getStore() }>
```
[the remark this answers is not in the document]
Comment (PreetiW) [reply src/AppEntry.js:12]: good catch @karelhala, I updated the code. 👍🏻

File: src/store/index.js (hunk 1 above)
Comment (PreetiW) [reply src/store/index.js:9]: Thank you so much @karelhala for the detailed explanation around it. This really helps me to understand more around it.
I wanted to know that is it okay if we go ahead with the current implementation done in the PR and lated introduce React.Context? will that still be fine?
also you mentioned ` when creating store use shortcutting` can you please tell me which shortcutting way you referring to?

Comment (karelhala) [reply src/store/index.js:9]: Yup, feel free to go with this approach to fix the imediate problem and refactor later on using `React.Context`.

As for the JS shortcutting/short-circuit you can read more about it here - https://codeburst.io/javascript-short-circuit-conditionals-bbc13ac3e9eb

**Review by karelhala — APPROVED** [review]

**Review by kgaikwad — APPROVED** [review]
Merging this pull request as original issue is fixed. 
As per discussion with @karelhala on PR, re-factoring part @PreetiW is going to address in a separate pull-request.

**Comment by kgaikwad** [comment]
Thank you @PreetiW for a fix!
Thank you @karelhala for details and review.

File: src/store/index.js (hunk 1 above)
Comment (PreetiW) [reply src/store/index.js:9]: Thanks @karelhala for the inputs 👍🏻

**Pull request merged**